src/main/java/org/passay/PasswordValidator.java

```diff
@@ -55,14 +59,114 @@ public PasswordValidator(final MessageResolver resolver, final List<Rule> rules)
   public RuleResult validate(final PasswordData passwordData)
   {
     final RuleResult result = new RuleResult(true);
-    for (Rule rule : passwordRules) {
-      final RuleResult rr = rule.validate(passwordData);
-      if (!rr.isValid()) {
-        result.setValid(false);
-        result.getDetails().addAll(rr.getDetails());
+    passwordRules.stream().map((rule) -> rule.validate(passwordData)).filter((rr) -> !rr.isValid()).forEach((rr) -> {
+      result.setValid(false);
+      result.getDetails().addAll(rr.getDetails());
+    });
+    return result;
+  }
+
+
+  /**
+   * Calculates the entropy of the given {@link PasswordData} based on the password rules specified in this instance.
+   * <p>
+   * <b>It's important to note that this method does NOT take into account {@link IllegalRegexRule} or
+   * {@link AllowedRegexRule}</b> as the regular expressions driving the rules may be negative matches.
+   *
+   * @param  passwordData  to validate
+   * @throws org.passay.EntropyException Thrown when passwordData {@link PasswordOrigin} is not implemented
+   * by this method or if the required {@link CharacterRule} instances are unavailable in the passwordRules of
+   * this validator instance.
+   *
+   * @see <a href="http://csrc.nist.gov/publications/nistpubs/800-63-1/SP-800-63-1.pdf">PDF Publication</a>
+   * @return  rule result
+   */
+  public Double getEntropyBits(final PasswordData passwordData)
```

Comment:
Well, I can totally see that.  I wanted to keep the entropy calculation native to the validator solely from the library user's perspective, and the fact that it can be considered a password utility (all static).  Feel free to merge separately, I personally wanted to put all the Rule implementations and the interface under a rule package, but thought that might be a major version.